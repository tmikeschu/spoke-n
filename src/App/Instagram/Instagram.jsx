import React, { Component } from "react"
import APIService from "../APIService/APIService"

const apiService = new APIService("https://spoken-api.herokuapp.com")

export default class Instagram extends Component {
  componentDidMount() {
    this.props.actions.fetchPhotos(apiService)
  }

  render() {
    const photos = this.props.photos.map((photo, i) => {
      return (
        <div key={i} >
          <a href={photo.link} target="_blank" rel="noopener noreferrer">
            <img src={photo.image} alt={photo.caption}/>
          </a>
          <section>
            <article></article>
            <p>"{photo.caption}"</p>
            <p>(click image to see on Instagram)</p>
          </section>
        </div>
      )
    })

    return (
      <div className="instagram">
        <article className="photos" >
          <section>
            <h3>See Our Journey</h3>
            <a href="https://www.instagram.com/thespokentour/" target="_blank" rel="noopener noreferrer">Follow us!<img src="http://www.underconsideration.com/brandnew/archives/instagram_2016_icon.jpg" alt="instagram"/></a>
          </section>
          { photos }
        </article>
      </div>
    )
  }
}
