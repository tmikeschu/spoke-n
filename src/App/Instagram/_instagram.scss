.instagram {
  background-color: #fff;
  border: 1px dotted $brown;
  max-width: 1000px;
  margin: 1% auto;
  padding-bottom: 4%;
  display: flex;
  flex-direction: column;
  background-image: url("../../public/loading.gif");
  background-repeat: no-repeat;
  background-position: 50% 50%;
  background-size: 20%;
  .photos {
    width: 100%;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    overflow: scroll;
    section {
      display: flex;
      align-items: center;
      width: 90%;
      margin: 2% auto;
      justify-content: space-between;
      h3 {
        color: $green;
        text-shadow: 1px 1px $lilac;
      }
      a {
        display: flex;
        align-items: center;
        img {
          width: 50px;
        }
      }
    }
    div {
      margin: 1% auto;
      width: 90%;
      justify-content: center;
      display: flex;
      align-items: center;
      padding: 1%;
      a {
        img {
          width: 100%;
          max-width: 640px;
          border: 1px solid black;
        }
      }
      &:nth-of-type(even) {
        background-color: #fff;
        border: 1px solid #000;
       a {
         order: 2;
       }
       section {
        order: 1
       }
      }
      &:nth-of-type(odd) {
        background-color: rgb(0,0,0);
        color: white;
        a {
          order: 1;
        }
        section {
          order: 2;
        }
      }
      section {
        display: flex;
        width: 50%;
        flex-direction: column;
        justify-content: space-between;
        align-self: stretch;
        align-items: flex-start;
        p {
          margin: 5%;
          &:last-of-type {
            font-size: 1em;
            color: $yellow;
            text-shadow: none;
          }
        }
      }
    }
  }
  @media (max-width: 640px) {
    .photos {
      width: 100%;
      section {
        flex-direction: column;
        align-items: flex-start;
      }
      div {
        flex-direction: column;
        a {
          order: 1;
          img {
            width: 100%;
          }
        }
        section {
          order: 2;
          width: 100%;
          p {
            text-align: left;
            padding: 1px;
          }
        }
      }
    }
  }
}
