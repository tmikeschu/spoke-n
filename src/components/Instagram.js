import React, { Component } from 'react';
import '../stylesheets/App.css';

export default class Instagram extends Component {
  render() {
    return (
      <div className="instagram">
        <h2>Instagram</h2>
        <article className="photos">

        </article>
      </div>
    );
  }
}
