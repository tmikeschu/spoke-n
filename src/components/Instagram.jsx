import React, { Component } from 'react';
import $ from 'jquery';
import '../stylesheets/App.css';

export default class Instagram extends Component {
  constructor(props) {
    super(props);
    this.state = {
      instagramPhotos: [],
    }
  }

  componentDidMount() {
    var self = this;
    $.ajax({
      url: "http://spoken-api.herokuapp.com/api/v1/instagram_photos?api_key="+process.env.REACT_APP_RAILS_KEY,
      method: "GET",
    }).done(function(response) {
      self.setState({
        instagramPhotos: response
      })
    }).fail(function(error) {
      console.error("No");
    });
  }


  render() {
    const photos = this.state.instagramPhotos.map((photo, i) => {
      return (
        <div key={i} >
          <a href={photo.link}>
            <img src={photo.image} alt={photo.caption}/>
          </a>
          <section>
            <article></article>
            <p>"{photo.caption}"</p>
            <p>click image to see on Instagram</p>
          </section>
        </div>
      )
    });

    return (
      <div className="instagram">
        <a href="https://www.instagram.com/thespokentour/">Follow us!<img src="http://www.underconsideration.com/brandnew/archives/instagram_2016_icon.jpg" alt="instagram"/></a>
        <article className="photos">
          <h3>See Our Journey</h3>
          { photos }
        </article>
      </div>
    );
  }
}
