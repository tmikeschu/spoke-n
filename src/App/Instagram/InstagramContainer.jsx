import { connect } from "react-redux"
import { bindActionCreators } from "redux"
import * as actions from "../../Redux/actions"
import Instagram from "./Instagram"

const mapStateToProps = ({ photos }) => ({ photos })

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(Instagram)

