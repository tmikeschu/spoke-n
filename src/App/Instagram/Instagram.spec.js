import React, { Component } from 'react'
import { shallow, mount } from 'enzyme'
import Instagram from './Instagram'

const fakeResponse = {
  status: 201,
  data: ["AHH!", "YEAHH!"]
}

const fakeService = {
  get(url) {
    return fakeResponse
  }
}

describe('<Instagram />', () => {
  const insta = shallow(<Instagram />)

  it('renders without crashing', () => {
    expect(insta).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(insta.getNodes()).toMatchSnapshot()
    })
  })

  it('renders a heading, a link, and photos', () => {
    const photos = [{ image: '', caption: ''}]
    insta.setState({instagramPhotos: photos})

    expect(insta.find('h3').length).toEqual(1)
    expect(insta.find('section a').length).toEqual(1)
    expect(insta.find('.photos div').length).toEqual(1)
  })

  describe("#getPhotos", () => {
    it("is called upon mount", () => {
      const restore = Instagram.prototype.getPhotos
      const mock = Instagram.prototype.getPhotos = jest.fn()
      const insta = mount(<Instagram />)
      expect(mock).toHaveBeenCalled()

      Instagram.prototype.getPhotos = restore
    })

    it("it returns a response", async () => {
      const insta = shallow(<Instagram />).instance()
      const response = await insta.getPhotos(fakeService)

      expect(response).toMatchObject(fakeResponse)
    })

    it("calls #setPhotos", async () => {
      const insta = shallow(<Instagram />).instance()
      const spy = jest.spyOn(insta, "setPhotos")
      await insta.getPhotos(fakeService)

      expect(spy).toHaveBeenCalled()
      spy.mockRestore()
    })
  })

  describe("#setPhotos", () => {
    it("updates photo state", () => {
      const insta = shallow(<Instagram />).instance()
      expect(insta.state.instagramPhotos).toEqual([])

      insta.setPhotos(fakeResponse.data)

      expect(insta.state.instagramPhotos).toEqual(["AHH!", "YEAHH!"])
    })

    it("can take bad data", () => {
      const insta = shallow(<Instagram />).instance()
      expect(insta.state.instagramPhotos).toEqual([])

      insta.setPhotos(null)
      expect(insta.state.instagramPhotos).toEqual([])

      insta.setPhotos(undefined)
      expect(insta.state.instagramPhotos).toEqual([])
    })
  })
})
