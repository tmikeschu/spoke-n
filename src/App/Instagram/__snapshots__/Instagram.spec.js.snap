// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instagram /> snapshot is valid 1`] = `
Array [
  <div
    className="instagram"
>
    <article
        className="photos"
    >
        <section>
            <h3>
                See Our Journey
            </h3>
            <a
                href="https://www.instagram.com/thespokentour/"
                rel="noopener noreferrer"
                target="_blank"
            >
                Follow us!
                <img
                    alt="instagram"
                    src="http://www.underconsideration.com/brandnew/archives/instagram_2016_icon.jpg"
                />
            </a>
        </section>
    </article>
</div>,
]
`;
